tests/test_serialPort.cpp

```diff
@@ -1,120 +1,103 @@
 // SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
 // SPDX-License-Identifier: LGPL-3.0-or-later
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE SerialPortTest
+#include <boost/test/unit_test.hpp>
+using namespace boost::unit_test_framework;
+
 #include "DummyServer.h"
 #include "SerialCommandHandler.h"
-#include "SerialPort.h"
-#include "stringUtils.h"
 
-#include <iostream>
 #include <string>
-#include <vector>
 
-int main() {
+struct SerialCommandHandlerFixture {
   DummyServer dummy;
+  SerialCommandHandler s;
 
-  int testCode = 0;
-  std::string cmd;
-  std::string res;
+  SerialCommandHandlerFixture() : s(dummy.deviceNode) {}
+};
 
-  SerialCommandHandler s(dummy.deviceNode);
+// Declares and registers a fixture used by all test cases under a test suite.
+// Each test case in the subtree of the test suite uses the fixture.
+BOOST_FIXTURE_TEST_SUITE(SerialCommandHandlerTests, SerialCommandHandlerFixture)
 
-  // ****************************************************************************************************************
-  testCode = 0; // TEST 0
+BOOST_AUTO_TEST_CASE(testSimpleReply) {
   for(long l = 0; l < 10; l++) {
-    cmd = "reply" + std::to_string(l);
+    std::string cmd = "reply" + std::to_string(l);
     s.write(cmd);
-    res = s.waitAndReadline(); // able to receive just fine after sending.
-    if(res != cmd) {
-      return std::max(1, testCode + ((int)l) + 1);
-    }
+    std::string res = s.waitAndReadline();
+    BOOST_TEST(res == cmd, "Failed simple reply test for iteration " << l);
```

Comment:
Providing a custom failure string here is less useful than the default printout, which also prints the data content of the failed comparison. As the loop counter is also in the test string, there is no added value here.
```suggestion
    BOOST_TEST(res == cmd);
```
Note: do a permutation test and change the line to ```BOOST_TEST(res == cmd+"_")``` and compare it to the output of BOOST_TEST(res == cmd+"_", "Failed simple reply test for iteration " << l);